libs/subprojects.settings.gradle

```
@@ -1,21 +1,20 @@
 // Copyright 2020 The Terasology Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-// This magically allows subdirs to become included builds
-// https://docs.gradle.org/6.4.1/userguide/composite_builds.html
-file(".").eachDir { possibleIncludedBuildDirectory ->
-    File buildFile = new File(possibleIncludedBuildDirectory, "build.gradle")
-    File settingsFile = new File(possibleIncludedBuildDirectory, "settings.gradle")
-
-    if (buildFile.exists() && settingsFile.exists()) {
-        logger.info("{} will be included in the composite build.",
-            rootDir.relativePath(possibleIncludedBuildDirectory))
-        includeBuild(possibleIncludedBuildDirectory)
-    } else {
-        logger.warn("{} REJECTED as an included build. build.gradle: {}, settings.gradle: {}",
-                rootDir.relativePath(possibleIncludedBuildDirectory),
-                buildFile.exists() ? "present" : "MISSING",
-                settingsFile.exists() ? "present" : "MISSING"
-        )
+// This magically allows subdirs in this subproject to themselves become sub-subprojects in a proper tree structure
+new File(rootDir, 'libs').eachDir { possibleSubprojectDir ->
+    def subprojectName = ':libs:' + possibleSubprojectDir.name
+    File buildFile = new File(possibleSubprojectDir, "build.gradle")
+    File settingsFile = new File(possibleSubprojectDir, "settings.gradle")
+    if (!buildFile.exists()) {
+        logger.warn("***** WARNING: Found a lib without a build.gradle, corrupt dir? NOT including {} *****", subprojectName)
+        return
+    }
+    if (!settingsFile.exists()) {
+        logger.warn("lib {} has build.gradle, but no settings.gradle? NOT including it." subprojectName)
+        return
     }
+    logger.info("lib {} has a build file so counting it complete and including it.", subprojectName)
+    includeBuild(subprojectName)
```

```suggestion
// This magically allows subdirs in this subproject to themselves become sub-subprojects in a proper tree structure
new File(rootDir, 'libs').eachDir { possibleSubprojectDir ->
    def subprojectName = ':libs:' + possibleSubprojectDir.name
    File buildFile = new File(possibleSubprojectDir, "build.gradle")
    File settingsFile = new File(possibleSubprojectDir, "settings.gradle")
    if (!buildFile.exists()) {
        logger.warn("***** WARNING: Found a lib without a build.gradle, corrupt dir? NOT including {} *****", subprojectName)
        return
    }
    if (!settingsFile.exists()) {
        logger.warn("lib {} has build.gradle, but no settings.gradle? NOT including it." subprojectName)
        return
    }
    logger.info("lib {} has a build file so counting it complete and including it.", subprojectName)
    includeBuild(possibleSubprojectDir)
```

`includeBuild` takes-in a file, not a file path. These changes do not actually work without it.

You can test this by doing:
 - `./groovyw lib get TeraNUI`
 - `./gradlew :TeraNUI:build`

---

added this as a comment so one sees this when looking at the source code already.